mkosi/__init__.py

```diff
@@ -2144,66 +2148,82 @@ def maybe_compress(
                 run(cmd, stdin=i, stdout=o, sandbox=context.sandbox(binary=cmd[0]))
 
 
-def copy_uki(context: Context) -> None:
-    if (context.staging / context.config.output_split_uki).exists():
-        return
+def copy_nspawn_settings(context: Context) -> None:
+    if context.config.nspawn_settings is None:
+        return None
+
+    with complete_step("Copying nspawn settings file…"):
+        shutil.copy2(context.config.nspawn_settings, context.staging / context.config.output_nspawn_settings)
+
 
+def get_uki(context: Context) -> Optional[Path]:
     if not want_efi(context.config) or context.config.unified_kernel_images == ConfigFeature.disabled:
-        return
+        return None
+
+    uki = context.root / efi_boot_binary(context)
+    if uki.exists() and (KernelType.identify(context.config, uki) == KernelType.uki):
```

Comment:
when moving code, please keep the style in tact (within reason), since that makes reviewing the diff easier, since git can then colour the move as a move instead of as removed and added lines.
```suggestion
    if (uki := context.root / efi_boot_binary(context)).exists() and (
        KernelType.identify(context.config, uki) == KernelType.uki
    ):
```